### ISSUE TITLE
solveset_complex fails on -I*(exp(I*x) - exp(-I*x))/2 - 1

### ISSUE THREAD

**Issue by hargup** [sympy/sympy#9759]
`solveset_complex` is not able to see the expression as a rational function in `exp(I*x)`. Some tweeking in `_solve_as_poly` should fix it. And as once Aaron pointed out in a mailing list discussion we should implement a solver based on composition of functions.

**Comment by Shekharrajak** [comment]
What output is expected ?

**Comment by kshitij10496** [comment]
```python
In []: solveset_complex(-I*(exp(I*x) - exp(-I*x))/2 - 1, x)
Out[]: 
⎧        π        ⎫
⎨2⋅n⋅π + ─ | n ∊ ℤ⎬
⎩        2        ⎭
```
The code for `solveset_complex` has undergone some behavioural since the time of creation of this issue.
And I think now we get the expected output. Isn't it so ?
@hargup

**Comment by hargup** [comment]
> And I think now we get the expected output. Isn't it so ?

Yeah, it works fine now. It was probably fixed in #10448, can you do a `git bisect` and confirm?

**Comment by kshitij10496** [comment]
Cool 
Thanks for the tip
